api/serializers.py

```diff
@@ -0,0 +1,76 @@
+from rest_framework import serializers
+from updates.models import Host
+from hwdoc.models import Equipment
+
+
+class EquipmentSerializer(serializers.HyperlinkedModelSerializer):
+
+    project = serializers.SerializerMethodField('get_allocation')
+    rack = serializers.SerializerMethodField('get_rack')
+    ipmi = serializers.SerializerMethodField('get_ipmi')
+
+    # used to get value from 'allocation' field.
+    # We need this to be able to display 'allocation'
+    # with a different name
+    def get_allocation(self, obj):
+        return obj.allocation
+
+    def get_rack(self, obj):
+        try:
+            rack = obj.rack.name
+            unit = obj.unit
+        except Exception:
+            rack = None
+            unit = None
+        return {
+            'rack': rack,
+            'rackunit': unit,
+        }
+
+    def get_ipmi(self, obj):
+        try:
+            host = obj.servermanagement.hostname
+            ip = obj.servermanagement.mac
+        except Exception:
+            host = None
+            ip = None
+        return {
+            'IPMI Hostname': host,
+            'IPMI Mac': ip
+        }
+
+    class Meta:
+        model = Equipment
+        fields = (
+            'serial',
+            'project',
+            'rack',
+            'ipmi',
+        )
+
+
+class HostSerializer(serializers.HyperlinkedModelSerializer):
+
+    equipment = serializers.SerializerMethodField('get_equipment')
+    hostname = serializers.SerializerMethodField('get_hostname')
+
+    def get_hostname(self, obj):
+        return obj.name
+
+    def get_equipment(self, obj):
+        try:
+            ret = Equipment.objects.get(
+                serial=obj.get_fact_value('serialnumber')
+            )
+            # import pdb; pdb.set_trace()
+            ret = EquipmentSerializer(ret).data
+        except Exception:
+            ret = None
+        return ret
+
+    class Meta:
+        model = Host
+        fields = (
```

Comment:
if you look at this, it's 1 fields from Host and all other fields are from Equipment. And I am not sure why we 'd want to expose any Host data yet. But even if we want, the Equipment is the one with all the data, it rightfully belongs to be the called one and not just be a child of Host.